### ISSUE TITLE
Unexpected errors when accessing attributes on variables named "match"

### ISSUE THREAD

**Issue by ahem** [microsoft/pyright#1492]
**Describe the bug**
Using pyright 1.1.111 I have started seeing errors when accessing properties on variables named `match`.

**To Reproduce**

This code generates error:
```python
from dataclasses import dataclass
from typing import List

@dataclass
class Match:
    list: List[str]

match = Match([])
match.list.append("wat")
```

Pyright output:
```
ahm@freezevision ~ % npx pyright file.py
No configuration file found.
stubPath /Users/ahm/typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
/Users/ahm/file.py
  9:1 - error: Match statements require Python 3.10 or newer
  9:6 - error: Expected expression
  9:25 - error: Expected ":"
3 errors, 0 warnings, 0 infos
Completed in 0.574sec
```

**Expected behavior**

Match expressions are purposely introduced as ["soft" keywords](https://www.python.org/dev/peps/pep-0622/#use-a-hard-keyword) to avoid causing this sort of errors in existing code, so PyRight should not flag this code as an error.

**VS Code extension or command-line**
```
ahm@freezevision ~ % npx pyright --version
pyright 1.1.111
```

**Comment by erictraut** [comment]
Thanks for reporting this, and apologies for the regression. I had added logic to handle "match" as a soft keyword, but it wasn't comprehensive, and there were some cases that I missed. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.112, which I just published. It will also be included in the next release of Pylance.
